### PR TITLE
Introduce a way to disable Abseil library

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**:

Create build switch and add a compiler definition `DISABLE_ABSEIL` to make a custom build possible to build w/o abseil library.

**Motivation and Context**

### CODE REVIEW

**Comment by snnn** [comment]
Sadly I was hoping to use more absl features.

**Review by pranavsharma — APPROVED** [review]

**Comment by pranavsharma** [comment]
> Sadly I was hoping to use more absl features.

We should continue to use abseil as this build flag is off by default. It's being introduced for one specific customer.

File: tools/ci_build/build.py [hunk 1]
```diff
@@ -580,6 +580,7 @@ def convert_arg_line_to_args(self, arg_line):
     )
     # Please note in our CMakeLists.txt this is already default on. But in this file we reverse it to default OFF.
     parser.add_argument("--disable_rtti", action="store_true", help="Disable RTTI (reduces binary size)")
+    parser.add_argument("--disable_abseil", action='store_true', help="Do no use abseil and default to STL")
     parser.add_argument(
```
Comment (skottmckay) [inline tools/ci_build/build.py:584]: For obscure things like can they specify via cmake_extra_args so we're not adding excessive amounts of new parameters to build.py

File: onnxruntime/core/framework/session_state.h [hunk 2]
```diff
@@ -503,7 +503,11 @@ class SessionState {
   PrepackedWeightsContainer* const prepacked_weights_container_{};
 
 #if !defined(ORT_MINIMAL_BUILD)
+#ifndef DISABLE_ABSEIL
   InlinedHashMap<InlinedVector<int>, InlinedHashSet<NodeIndex>> to_be_executed_nodes_;
+#else
+  std::map<InlinedVector<int>, InlinedHashSet<NodeIndex>> to_be_executed_nodes_;
```
Comment (edgchen1) [inline onnxruntime/core/framework/session_state.h:509]: why did this change from InlinedHashMap to std::map?

**Review by skottmckay — APPROVED** [review]

File: onnxruntime/core/framework/session_state.h (hunk 2 above)
Comment (yuslepukhin) [reply onnxruntime/core/framework/session_state.h:509]: std::unordered_map does not accept std::vector as a key. I reverted it to std::map, which was the original type before refactoring.

**Comment by yuslepukhin** [comment]
> > Sadly I was hoping to use more absl features.
> 
> We should continue to use abseil as this build flag is off by default. It's being introduced for one specific customer.
> 
> In reply to: [1110281199](https://github.com/microsoft/onnxruntime/pull/11353#issuecomment-1110281199)

This constraints us to using typedefs at the `inlined_containers.h` and not a free use of `absl::` namespace.

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**